modules/utxo-lib/src/noble_ecc.ts

```diff
@@ -0,0 +1,95 @@
+import * as createHash from 'create-hash';
+import * as createHmac from 'create-hmac';
+import { ECPairAPI, ECPairFactory } from 'ecpair';
+import * as necc from '@noble/secp256k1';
+import { BIP32API, BIP32Factory } from 'bip32';
+
+necc.utils.sha256Sync = (...messages: Uint8Array[]): Uint8Array => {
+  const sha256 = createHash('sha256');
+  for (const message of messages) sha256.update(message);
+  return sha256.digest();
+};
+
+necc.utils.hmacSha256Sync = (key: Uint8Array, ...messages: Uint8Array[]): Uint8Array => {
+  const hash = createHmac('sha256', Buffer.from(key));
+  messages.forEach((m) => hash.update(m));
+  return Uint8Array.from(hash.digest());
+};
+
+const defaultTrue = (param?: boolean): boolean => param !== false;
+
+function throwToNull<Type>(fn: () => Type): Type | null {
+  try {
+    return fn();
+  } catch (e) {
+    return null;
+  }
+}
+
+function isPoint(p: Uint8Array, xOnly: boolean): boolean {
+  if ((p.length === 32) !== xOnly) return false;
+  try {
+    return !!necc.Point.fromHex(p);
+  } catch (e) {
+    return false;
+  }
+}
+
+const ecc = {
```

Comment:
If this is going to be used across different coins types (eth, utxo, etc), it would seem like it would be better served not being exported from `utxo-lib`.

Reply:
Great idea! Where do you suggest?

Reply:
It would seem like `sdk-core` would be better than `abstract-utxo` since the implementation is coin agnostic.

Reply:
Moving it to sdk-core creates a circular dependency between `utxo-lib` and `sdk-core` :cry:. Might have to live here for now.

Reply:
Ah biscuits...